WalletWasabi.Tests/UnitTests/Tor/TorBinaryHashesTests.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Threading;
+using WalletWasabi.Microservices;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.Tor
+{
+	public class TorBinaryHashesTests
+	{
+		[Fact]
+		public void VerifyTorBinaryChecksumHashes()
```

Comment:
The same as above.

Reply:
I didn't find a file to use to check against those hashes.

Reply:
This should be documented too. One should know where to get those binaries.

Reply:
@molnard Do you know where to get those binaries?

Reply:
Yes, I download the Tor browser and gather the binaries from it, manually. https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Documentation/Guides/UpgradeTorDaemons.md

Reply:
Ok, but where can I find something similar to https://bitcoinknots.org/files/0.20.x/0.20.0.knots20200614/SHA256SUMS.asc for our current Tor version?

Reply:
You probably won't. There is:

![image](https://user-images.githubusercontent.com/58662979/104583309-4555cf00-5661-11eb-9aba-2a9071f45431.png)

but that's not a hash. So whoever wants to verify our hashes has to download Tor from https://www.torproject.org/download/, verify that the file is correct. Extract archive/msi and verify that hash of the binary is the same as ours.

Reply:
I am not sure what should be documented and where can you get older Tor versions?

Reply:
> where can you get older Tor versions?

I can't find it but typically you upgrade to the latest version. So it's not ideal but it's what it is.

> I am not sure what should be documented

Well, you just add some instructions how to verify that those hashes are correct. Nothing fancy.

Reply:
> I can't find it but typically you upgrade to the latest version. So it's not ideal but it's what it is.

But if the ones who want to verify our hashes can't find our old Tor version how can they verify them?

Reply:
There is file hippo and other services. However, this looks good https://dist.torproject.org/ & https://dist.torproject.org/torbrowser/

Reply:
I cannot find older Tor versions in those links.

Reply:
You cannot find it. Isn't it good like this? Developer updates the Tor binaries and ensures they are valid. Run the tests it will fail and update the Tests with the new hashes.

Reply:
It's great as it is. With older Tor releases, it would be top notch, but when they are not available, we are out of luck.